heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraScheduler.java

```diff
@@ -190,6 +194,22 @@ protected String getHeronAuroraPath() {
     return auroraProperties;
   }
 
+  /**
+   * This method finds the container with highest resource requirement and returns the resource.
+   * Currently only RAM is used for max identification.
+   */
+  private Resource getMaxRequiredResource(PackingPlan packingPlan) {
```

Comment:
I like the first suggestion, some helper method like `getHomogeneousResource`. As I understand, the helper method will be used by a scheduler to update the `PackingPlan`, so that downstream components are up-to-date with the actual allocation. 

Regarding your second point, CPU and Disk need similar treatment. However a couple of improvements required in RoundRobinPacking first. CPU and disk do not have the same api and config support as RAM. It could have been done to reduce complexity. I would suggest that CPU and disk be discussed in a separate thread.